quickjs-libc.c

```diff
@@ -2051,7 +2046,7 @@ static JSValue js_os_setTimeout(JSContext *ctx, JSValueConst this_val,
         return JS_EXCEPTION;
     }
     th->has_object = TRUE;
-    th->timeout = get_time_ms() + delay;
+    th->timeout = (js__hrtime_ns() / 1e6) + delay;
```

Comment:
Just a suggestion because I know some people like parentheses for clarity but... I'm not one of those people :)
```suggestion
    th->timeout = js__hrtime_ns() / 1e6 + delay;
```